### DISCUSSION TITLE
Support for validation by some Editor/IDE, and/or generation of YAML-Schema / JSON-Schema?

### DISCUSSION THREAD

**Discussion by alf63** [omry/omegaconf#1170] [Q&A]
Hello,

My need is to **validate my YAML files against their schema defined in Python/dataclasses/OmegaConf, in some Editor/IDE (assistance to edition)**.
I have no YAML-Schema, as **the only reference "schema"** is the one in **.py**.

So I have two complementary questions:

**(1)** Is there any (planned?) support, in some Editors/IDEs, for **validation of YAML files directly against .py dataclasses** (the ones used by OmegaConf to parse the YAML)?

_or, by default, as several editors/IDEs support validation of YAML files against YAML-Schema (or even JSON-Schema):_

**(2)** Is there any (planned?) support, in OmegaConf, for **generation of YAML-Schema and/or JSON-Schema from the .py files**?
(or do you know of any such support, elsewhere than in OmegaConf, as the format of the .py files is not really specific to OmegaConf...?)

Best regards

**Comment by omry** [comment]
At this point, IDE integration is out of scope for OmegaConf/Hydra.
